src/stages/scan-files.ts

```diff
@@ -1,112 +1,80 @@
-    const imports = root.findAll({
-      rule: {
-        any: [
-          {
-            pattern: {
-              context: `import $NAME from '${replacement.moduleName}'`,
-              strictness: 'relaxed'
-            }
-          },
-          {
-            pattern: {
-              context: `require('${replacement.moduleName}')`,
-              strictness: 'relaxed'
-            }
-          }
-        ]
-      }
-    });
-
-    if (imports.length > 0) {
-      matches.push(replacement);
-    }
-
-    for (const node of imports) {
-      const range = node.range();
-      let snippet: string = '';
-
-      const prevLine = lines[range.start.line - 1];
-      const line = lines[range.start.line];
-      const nextLine = lines[range.start.line + 1];
-
-      if (prevLine) {
-        snippet += `${range.start.line} | ${prevLine}\n`;
-      }
-
-      snippet += `${range.start.line + 1} | ${pc.red(line)}\n`;
+const events = new Events();
 
-      if (nextLine) {
-        snippet += `${range.start.line + 2} | ${nextLine}\n`;
-      }
+let tasks = 0;
 
-      suggestReplacement(replacement, {
-        type: 'file',
-        path: filePath,
-        line: range.start.line,
-        column: range.start.column,
-        snippet
-      });
+export async function _scanFile(
+  file: string,
+  replacements: modReplacements.ModuleReplacement[]
+): Promise<FileReplacement> {
+  // return new Promise((resolve, reject) => {
+  const worker = fork(`${__dirname}/workers/scan-file.js`);
+  worker.on('error', (error) => reject(error));
+  worker.once('message', (message) => {
+    if (message?.type === 'result') {
+      // resolve(message.value);
+      events.emit('file-scan-worker-result', message.value);
+    } else {
+      console.error(message.value);
+
+      // reject(message.value);
     }
-  }
-
-  if (matches.length === 0) {
-    return null;
-  }
-
-  return {
-    path: filePath,
-    contents,
-    replacements: matches
-  };
+    events.emit('file-scan-worker-done');
+    worker.kill();
+  });
+  worker.send({file, replacements});
+  // });
 }
 
-export async function scanFiles(
+export function scanFiles(
   files: string[],
   replacements: modReplacements.ModuleReplacement[],
-  spinner: ReturnType<typeof cl.spinner>
+  spinner: ReturnType<typeof cl.spinner>,
+  results: FileReplacement[]
 ): Promise<FileReplacement[]> {
-  const results: FileReplacement[] = [];
-
-  for (const file of files) {
-    try {
-      const contents = await readFile(file, 'utf8');
-      const lines = contents.split('\n');
-
-      spinner.message(`Scanning ${file}`);
-
-      const scanResult = await scanFile(file, contents, lines, replacements);
+  return new Promise((resolve, reject) => {
+    if (!results) results = [];
+    let i = 0;
+    const filesLength = files.length;
+
+    const runJob = () => {
+      const available = availableParallelism();
+
+      let maxThreads;
+      if (!tasks) maxThreads = available;
```

Comment:
nit: run the formatter and if it doesn't bin these one-line ifs, do it yourself (for consistency in the repo)